pkg/rabbitmq/cluster.go

```diff
@@ -0,0 +1,284 @@
+			Name:  "PATH",
+			Value: "/usr/lib/rabbitmq/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
+		},
+	}
+
+	inetFamily := "inet"
+	inetProtocol := "tcp"
+	tlsArgs := ""
+	fipsArgs := ""
+	if IPv6Enabled {
+		inetFamily = "inet6"
+	}
+	erlangInetConfig := fmt.Sprintf("{%s,true}.\n", inetFamily)
+
+	if cluster.Spec.TLS.SecretName != "" {
+		inetProtocol = "tls"
+		tlsArgs = "-ssl_dist_optfile /etc/rabbitmq/inter-node-tls.config"
+		if fipsEnabled {
+			fipsArgs = "-crypto fips_mode true"
+		}
+	}
+
+	envVars = append(envVars, corev1.EnvVar{
+		Name: "RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS",
+		Value: fmt.Sprintf(
+			"-kernel inetrc '/etc/rabbitmq/erl_inetrc' -proto_dist %s_%s %s %s",
+			inetFamily,
+			inetProtocol,
+			tlsArgs,
+			fipsArgs,
+		),
+	}, corev1.EnvVar{
+		Name:  "RABBITMQ_CTL_ERL_ARGS",
+		Value: fmt.Sprintf("-proto_dist %s_%s %s", inetFamily, inetProtocol, tlsArgs),
+	})
+
+	defaultStatefulSet := rabbitmqv2.StatefulSet{
+		Spec: &rabbitmqv2.StatefulSetSpec{
+			Template: &rabbitmqv2.PodTemplateSpec{
+				EmbeddedObjectMeta: &rabbitmqv2.EmbeddedObjectMeta{},
+				Spec: &corev1.PodSpec{
+					SecurityContext: &corev1.PodSecurityContext{},
+					Containers: []corev1.Container{
+						{
+							// NOTE(gibi): if this is set according to the
+							// RabbitMQCluster name the the Pod will crash
+							Name:  "rabbitmq",
+							Image: cluster.Spec.Image,
+							Env:   envVars,
+							Args: []string{
+								// OSP17 runs kolla_start here, instead just run rabbitmq-server directly
+								"/usr/lib/rabbitmq/bin/rabbitmq-server",
+							},
+						},
+					},
+					InitContainers: []corev1.Container{
+						{Name: "setup-container", SecurityContext: &corev1.SecurityContext{}}},
+				},
+			},
+		},
+	}
+
+	if cluster.Spec.Override.StatefulSet == nil {
+		cluster.Spec.Override.StatefulSet = &defaultStatefulSet
+	}
+
+	if cluster.Spec.Override.StatefulSet.Spec.Template.Spec.NodeSelector == nil {
+		if nodeselector != nil {
+			cluster.Spec.Override.StatefulSet.Spec.Template.Spec.NodeSelector = *nodeselector
+		}
+	}
+
+	if topology != nil {
+		// Get the Topology .Spec
+		ts := topology.Spec
+		// Process TopologySpreadConstraints if defined in the referenced Topology
+		if ts.TopologySpreadConstraints != nil {
+			cluster.Spec.Override.StatefulSet.Spec.Template.Spec.TopologySpreadConstraints = *topology.Spec.TopologySpreadConstraints
+		}
+		// Process Affinity if defined in the referenced Topology
+		if ts.Affinity != nil {
+			cluster.Spec.Override.StatefulSet.Spec.Template.Spec.Affinity = ts.Affinity
+		}
+	}
+	if cluster.Spec.Affinity == nil {
+		cluster.Spec.Affinity = &corev1.Affinity{
+			PodAntiAffinity: &corev1.PodAntiAffinity{
+				PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{
+					{
+						Weight: 100,
+						PodAffinityTerm: corev1.PodAffinityTerm{
+							LabelSelector: &metav1.LabelSelector{
+								MatchExpressions: []metav1.LabelSelectorRequirement{
```

Comment:
if we create `serviceLabels` in the controller (that we use as `defaultLabelSelector` for topology), we could pass them down to the `StatefulSet` (but not sure it is possible for the `rabbitmqv2.StatefulSet`) we might have the same selectors here instead of `MatchExpression`, or do we have a different reason to only use `k8sAppName` ?

Reply:
I think I answered my own question looking at https://github.com/rabbitmq/cluster-operator/blob/main/api/v1beta1/rabbitmqcluster_types.go and I see

```
  selector:
    matchLabels:
      app.kubernetes.io/name: rabbitmq
```